packages/hop-node/src/watchers/BondTransferRootWatcher.ts

```diff
@@ -136,10 +95,21 @@ class BondTransferRootWatcher extends BaseWatcher {
       logger.debug(
         `transferRootHash ${transferRootHash} already bonded. skipping.`
       )
+      const event = await l1Bridge.getTransferRootBondedEvent(transferRootHash)
+      const { transactionHash } = event
+      const { from: sender } = await l1Bridge.getTransaction(
+        event.transactionHash
+      )
+      const timestamp = await this.bridge.getEventTimestamp(event)
+
       await this.db.transferRoots.update(transferRootHash, {
-        transferRootId,
         transferRootHash,
-        bonded: true
+        bonded: true,
+        bonder: sender,
+        bondTotalAmount: totalAmount,
+        bondTxHash: transactionHash,
```

Comment:
Yes. In the case where the syncer misses the event for some reason, we need a way to re-add the data to the DB so that the missed event is now "seen". This is similar to the logic we have in the BondWithdrawalWatcher [here](https://github.com/hop-protocol/hop/blob/develop/packages/hop-node/src/watchers/BondWithdrawalWatcher.ts#L101-L117).

I agree it should be DRYer, but it would take a slight architecture change because the watchers don't have access to the syncer functions. Ideally these lines would just be replaced with a `handleTransferRootBondedEvent()` call.